File: task-management-frontend/src/models/task.ts
export interface Task {
    id: number;
    title: string;
    description?: string;
    isCompleted: boolean;
    dueDate?: string;
    createdAt?: string;
    updatedAt?: string;
    userId?: string;
}

export interface TaskForm {
    id?: number;
    title: string;
    description?: string;
    dueDate?: string;
}

File: TaskManagementApi/Entities/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManagementApi.Entities
{
    public class Task 
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; }

        [MaxLength(1000)]
        public string? Description { get; set; }

        public bool IsCompleted { get; set; }

        public DateTime? DueDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public string UserId { get; set; }
    }
}